src/main/java/com/minecolonies/coremod/event/EventHandler.java

```diff
@@ -56,26 +70,110 @@ public void onDebugOverlay(final RenderGameOverlayEvent.Text event)
                 final WorldClient world = mc.world;
                 final EntityPlayerSP player = mc.player;
                 IColony colony = ColonyManager.getIColony(world, player.getPosition());
-                final double minDistance = ColonyManager.getMinimumDistanceBetweenTownHalls();
 
                 if (colony == null)
                 {
+                    if (!ColonyManager.isTooCloseToColony(world, player.getPosition()))
+                    {
+                        event.getLeft().add(LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.noCloseColony"));
+                        return;
+                    }
                     colony = ColonyManager.getClosestIColony(world, player.getPosition());
 
-                    if (colony == null || Math.sqrt(colony.getDistanceSquared(player.getPosition())) > 2 * minDistance)
+                    if (colony == null)
                     {
-                        event.getLeft().add(LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.noCloseColony"));
                         return;
                     }
 
                     event.getLeft().add(LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.nextColony",
-                      (int) Math.sqrt(colony.getDistanceSquared(player.getPosition())), minDistance));
+                            (int) Math.sqrt(colony.getDistanceSquared(player.getPosition())), ColonyManager.getMinimumDistanceBetweenTownHalls()));
                     return;
                 }
 
                 event.getLeft().add(colony.getName() + " : "
-                                      + LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.blocksFromCenter",
-                  (int) Math.sqrt(colony.getDistanceSquared(player.getPosition()))));
+                        + LanguageHandler.format("com.minecolonies.coremod.gui.debugScreen.blocksFromCenter",
+                        (int) Math.sqrt(colony.getDistanceSquared(player.getPosition()))));
+            }
+        }
+    }
+
+    /**
+     * Event called to attach capabilities.
+     * @param event the event.
+     */
+    @SubscribeEvent
+    public void onAttachingCapabilities(@NotNull final AttachCapabilitiesEvent<Chunk> event)
+    {
+        event.addCapability(new ResourceLocation(Constants.MOD_ID, "closeColony"), new ColonyTagCapabilityProvider());
+    }
+
+    /**
+     * Event called when the player enters a new chunk.
+     * @param event the event.
+     */
+    @SubscribeEvent
+    public void onEnteringChunk(@NotNull final PlayerEvent.EnteringChunk event)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 32 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)